yarn-project/simulator/src/private/unconstrained_execution_oracle.ts

```diff
@@ -279,20 +279,12 @@ export class UnconstrainedExecutionOracle extends TypedOracle {
   }
 
   public override async syncNotes() {
-    const taggedLogsByRecipient = await this.executionDataProvider.syncTaggedLogs(
+    await this.executionDataProvider.syncTaggedLogs(
       this.contractAddress,
       await this.executionDataProvider.getBlockNumber(),
       this.scopes,
     );
 
-    for (const [recipient, taggedLogs] of taggedLogsByRecipient.entries()) {
-      await this.executionDataProvider.processTaggedLogs(
-        this.contractAddress,
-        taggedLogs,
-        AztecAddress.fromString(recipient),
-      );
-    }
-
     await this.executionDataProvider.removeNullifiedNotes(this.contractAddress);
```

Comment:
@nventuro am I correct in thinking that this is a problem now because we now remove nullified notes before the logs are actually processed since at the point where `removeNullifiedNotes` is called the new logs are happily sitting in capsules waiting for processing instead of already being processed?

If you agree I guess we want to introduce a `removeNullifiedNotes` oracle and call that after we process all the log capsules.

Reply:
Got a DM from Nico that new notes are being checked for being nullified immediately so this is not a problem.